File: examples/python/hello_world/hello_world.py
#!/usr/bin/env python3
import argparse

from jinja2 import Environment, PackageLoader


def main():
    parser = argparse.ArgumentParser()
    parser.add_argument("--name", default="World", help="name to great")
    args = parser.parse_args()

    loader = PackageLoader("examples.python.hello_world", "templates")
    template = Environment(loader=loader).get_template("hello")
    print(template.render(name=args.name))


if __name__ == "__main__":
    main()

File: src/perl/hello_world/hello_world.pl
#!/usr/bin/perl
use strict;
use warnings;

use Getopt::Long;

sub greet {
    my ($name) = @_;
    return "Hello, $name!\n";
}

my $name = 'World';
GetOptions('name=s' => \$name);

print greet($name);
